### PR TITLE
[keymap] Adds tofu65 with split spacebar

### PR DESCRIPTION
**Pull request by bioe007**
## Description
This is for the KBDFans kbd67 rev2, but with a split spacebar. The keymap I've included is specifically for a tofu65 with three modifier keys between the rightmost spacebar and left-cursor key. 

## Types of Changes
Adds  LAYOUT_65_ansi_split_sp - macro to handle the additional keys created by splitting the spacebar.
Create keymap - ansi_split_space

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [ x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [ ] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [ x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [ ] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Review by drashna — APPROVED** [review]

File: keyboards/kbdfans/kbd67/rev2/keymaps/ansi_split_space/keymap.c [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+   * |Shift   |  Z|  X|  C|  V|  B|  N|  M|  ,|  .|  /|Shift | Up|End |
+   * |----------------------------------------------------------------|
+   * |Ctrl|Win |Alt | fn     |win| space    |Alt| FN|Ctrl|Lef|Dow|Rig |
+   * `----------------------------------------------------------------'
+   */
+[0] = LAYOUT_65_ansi_split_sp(
+  KC_GRV,  KC_1,    KC_2,   KC_3,   KC_4,   KC_5,   KC_6,   KC_7,   KC_8,    KC_9,    KC_0,    KC_MINS, KC_EQL,  KC_BSPC, KC_ESC,  \
+  KC_TAB,  KC_Q,    KC_W,   KC_E,   KC_R,   KC_T,   KC_Y,   KC_U,   KC_I,    KC_O,    KC_P,    KC_LBRC, KC_RBRC, KC_BSLS, KC_HOME, \
+  KC_LCTL, KC_A,    KC_S,   KC_D,   KC_F,   KC_G,   KC_H,   KC_J,   KC_K,    KC_L,    KC_SCLN, KC_QUOT,          KC_ENT, KC_PGUP,  \
+  KC_LSFT, KC_Z,    KC_X,   KC_C,   KC_V,   KC_B,   KC_N,   KC_M,   KC_COMM, KC_DOT,  KC_SLSH, KC_RSFT,           KC_UP, KC_PGDN,  \
```
Comment (fauxpark) [inline keyboards/kbdfans/kbd67/rev2/keymaps/ansi_split_space/keymap.c:42]:
```suggestion
[0] = LAYOUT_65_ansi_split_sp(
  KC_GRV,  KC_1,    KC_2,   KC_3,   KC_4,   KC_5,   KC_6,   KC_7,   KC_8,    KC_9,    KC_0,    KC_MINS, KC_EQL,  KC_BSPC, KC_ESC,
  KC_TAB,  KC_Q,    KC_W,   KC_E,   KC_R,   KC_T,   KC_Y,   KC_U,   KC_I,    KC_O,    KC_P,    KC_LBRC, KC_RBRC, KC_BSLS, KC_HOME,
  KC_LCTL, KC_A,    KC_S,   KC_D,   KC_F,   KC_G,   KC_H,   KC_J,   KC_K,    KC_L,    KC_SCLN, KC_QUOT,          KC_ENT, KC_PGUP,
  KC_LSFT, KC_Z,    KC_X,   KC_C,   KC_V,   KC_B,   KC_N,   KC_M,   KC_COMM, KC_DOT,  KC_SLSH, KC_RSFT,           KC_UP, KC_PGDN,
```

File: keyboards/kbdfans/kbd67/rev2/keymaps/ansi_split_space/keymap.c [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+   * |        |   |   |   |   |   |   |   |End|PDn|Dow|      |PUp|    |
+   * |----------------------------------------------------------------|
+   * |    |    |    |                       |   |   |    |Hom|PDn|End |
+   * `----------------------------------------------------------------'
+   */
+[1] = LAYOUT_65_ansi_split_sp(
+   KC_GRV,  KC_F1,  KC_F2,  KC_F3,  KC_F4,  KC_F5,  KC_F6,  KC_F7,  KC_F8,  KC_F9, KC_F10, KC_F11, KC_F12,          KC_DEL,KC_INS, \
+   _______,_______,  KC_UP,_______,_______,_______,_______,_______,KC_PSCR,KC_SLCK,KC_PAUS,  KC_UP,_______,     _______,_______,   \
+   _______,KC_LEFT,KC_DOWN,KC_RGHT,_______,_______,_______,_______,KC_HOME,KC_PGUP,KC_LEFT,KC_RGHT,            _______,_______, \
+   _______,_______,_______,_______,_______,_______,_______,_______, KC_END,KC_PGDN,KC_DOWN,       _______,KC_PGUP,_______, \
```
Comment (fauxpark) [inline keyboards/kbdfans/kbd67/rev2/keymaps/ansi_split_space/keymap.c:62]:
```suggestion
   KC_GRV,  KC_F1,  KC_F2,  KC_F3,  KC_F4,  KC_F5,  KC_F6,  KC_F7,  KC_F8,  KC_F9, KC_F10, KC_F11, KC_F12,          KC_DEL,KC_INS,
   _______,_______,  KC_UP,_______,_______,_______,_______,_______,KC_PSCR,KC_SLCK,KC_PAUS,  KC_UP,_______,     _______,_______,
   _______,KC_LEFT,KC_DOWN,KC_RGHT,_______,_______,_______,_______,KC_HOME,KC_PGUP,KC_LEFT,KC_RGHT,            _______,_______,
   _______,_______,_______,_______,_______,_______,_______,_______, KC_END,KC_PGDN,KC_DOWN,       _______,KC_PGUP,_______,
```

File: keyboards/kbdfans/kbd67/rev2/keymaps/ansi_split_space/keymap.c [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+
+}
+
+void matrix_scan_user(void) {
+
+}
+
+void led_set_user(uint8_t usb_led) {
+
+}
```
Comment (fauxpark) [inline keyboards/kbdfans/kbd67/rev2/keymaps/ansi_split_space/keymap.c:99]:
```suggestion
```

File: keyboards/kbdfans/kbd67/rev2/keymaps/ansi_split_space/keymap.c [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#include QMK_KEYBOARD_H
+
+// Defines the keycodes used by our macros in process_record_user
+enum custom_keycodes {
+  QMKBEST = SAFE_RANGE,
+  QMKURL
+};
+
```
Comment (fauxpark) [inline keyboards/kbdfans/kbd67/rev2/keymaps/ansi_split_space/keymap.c:23]:
```suggestion
```

File: keyboards/kbdfans/kbd67/rev2/keymaps/ansi_split_space/keymap.c [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+        // when keycode QMKURL is pressed
+        SEND_STRING("https://qmk.fm/" SS_TAP(X_ENTER));
+      } else {
+        // when keycode QMKURL is released
+      }
+      break;
+  }
+  return true;
+}
+
```
Comment (fauxpark) [inline keyboards/kbdfans/kbd67/rev2/keymaps/ansi_split_space/keymap.c:88]:
```suggestion
```

File: keyboards/kbdfans/kbd67/rev2/keymaps/ansi_split_space/config.h [hunk 6]
```diff
@@ -0,0 +1,19 @@
+/* Copyright 2018 MechMerlin
```
Comment (fauxpark) [inline keyboards/kbdfans/kbd67/rev2/keymaps/ansi_split_space/config.h:1]: This file can be deleted.

File: keyboards/kbdfans/kbd67/rev2/keymaps/ansi_split_space/readme.md [hunk 7]
```diff
@@ -0,0 +1,2 @@
+# Keyboard map for kbd67 rev.2 tofu65, with split spacebar,
```
Comment (fauxpark) [inline keyboards/kbdfans/kbd67/rev2/keymaps/ansi_split_space/readme.md:1]:
```suggestion
# Keymap for kbd67 rev.2 tofu65, with split spacebar
```

File: keyboards/kbdfans/kbd67/rev2/rev2.h [hunk 8]
```diff
@@ -101,3 +101,18 @@
     { K30, K31,   K32,   K33, K34,   K35,   K36, K37,   K38,   K39,   K3A, K3B, KC_NO, K3D,   K3E,   K3F }, \
     { K40, K41,   KC_NO, K43, KC_NO, KC_NO, K46, KC_NO, KC_NO, KC_NO, K4A, K4B, K4C,   K4D,   K4E,   K4F }, \
 }
+
+#define LAYOUT_65_ansi_split_sp( \
```
Comment (fauxpark) [inline keyboards/kbdfans/kbd67/rev2/rev2.h:105]:
```suggestion
#define LAYOUT_65_ansi_split_space( \
```

File: keyboards/kbdfans/kbd67/rev2/keymaps/ansi_split_space/keymap.c [hunk 9]
```diff
@@ -0,0 +1,99 @@
...
+   * |     |   |Up |   |   |   |   |   |PSc|SLk|Pau|Up |   |     |    |
+   * |----------------------------------------------------------------|
+   * |      |Lef|Dow|Rig|   |   |   |   |Hom|PUp|Lef|Rig|        |    |
+   * |----------------------------------------------------------------|
+   * |        |   |   |   |   |   |   |   |End|PDn|Dow|      |PUp|    |
+   * |----------------------------------------------------------------|
+   * |    |    |    |                       |   |   |    |Hom|PDn|End |
+   * `----------------------------------------------------------------'
+   */
+[1] = LAYOUT_65_ansi_split_sp(
```
Comment (fauxpark) [inline keyboards/kbdfans/kbd67/rev2/keymaps/ansi_split_space/keymap.c:58]:
```suggestion
[1] = LAYOUT_65_ansi_split_space(
```

File: keyboards/kbdfans/kbd67/rev2/keymaps/ansi_split_space/keymap.c [hunk 10]
```diff
@@ -0,0 +1,99 @@
...
+   * |Tab  |  Q|  W|  E|  R|  T|  Y|  U|  I|  O|  P|  [|  ]|  \  |PgUp|
+   * |----------------------------------------------------------------|
+   * |Caps   |  A|  S|  D|  F|  G|  H|  J|  K|  L|  ;|  '|Return |PgDn|
+   * |----------------------------------------------------------------|
+   * |Shift   |  Z|  X|  C|  V|  B|  N|  M|  ,|  .|  /|Shift | Up|End |
+   * |----------------------------------------------------------------|
+   * |Ctrl|Win |Alt | fn     |win| space    |Alt| FN|Ctrl|Lef|Dow|Rig |
+   * `----------------------------------------------------------------'
+   */
+[0] = LAYOUT_65_ansi_split_sp(
```
Comment (fauxpark) [inline keyboards/kbdfans/kbd67/rev2/keymaps/ansi_split_space/keymap.c:38]:
```suggestion
[0] = LAYOUT_65_ansi_split_space(
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/kbdfans/kbd67/rev2/keymaps/ansi_split_space/config.h (hunk 6 above)
Comment (bioe007) [reply keyboards/kbdfans/kbd67/rev2/keymaps/ansi_split_space/config.h:1]: Yeah, I thought it was odd but just sort of blindly following the pattern.

**Comment by bioe007** [comment]
Sorry for the slow response.. My current employer is not using GH and so I'm rarely here. I'll get this cleaned up now.

File: keyboards/kbdfans/kbd67/rev2/keymaps/ansi_split_space/keymap.c [hunk 11]
```diff
@@ -0,0 +1,81 @@
...
+      if (record->event.pressed) {
+        // when keycode QMKURL is pressed
+        SEND_STRING("https://qmk.fm/" SS_TAP(X_ENTER));
+      } else {
+        // when keycode QMKURL is released
+      }
+      break;
+  }
+  return true;
+}
```
Comment (fauxpark) [inline keyboards/kbdfans/kbd67/rev2/keymaps/ansi_split_space/keymap.c:81]:
```suggestion
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Review by fauxpark — APPROVED** [review]

**Pull request merged**